service/gcs/stdio/stdio.go

```diff
@@ -100,15 +105,37 @@ func (s *ConnectionSet) Files() (_ *FileSet, err error) {
 }
 
 // NewTtyRelay returns a new TTY relay for a given master PTY file.
-func (s *ConnectionSet) NewTtyRelay(pty io.ReadWriteCloser) *TtyRelay {
+func (s *ConnectionSet) NewTtyRelay(pty *os.File) *TtyRelay {
 	return &TtyRelay{s: s, pty: pty}
 }
 
 // TtyRelay relays IO between a set of stdio connections and a master PTY file.
 type TtyRelay struct {
-	wg  sync.WaitGroup
-	s   *ConnectionSet
-	pty io.ReadWriteCloser
+	closing int32
+	wg      sync.WaitGroup
+	s       *ConnectionSet
+	pty     *os.File
+}
+
+// ResizeConsole sends the appropriate resize to a pTTY FD
+func (r *TtyRelay) ResizeConsole(height, width uint16) error {
+	type consoleSize struct {
+		Height uint16
+		Width  uint16
+		x      uint16
+		y      uint16
+	}
+
+	r.wg.Add(1)
+	defer r.wg.Done()
+	if atomic.LoadInt32(&r.closing) != 0 {
+		return errors.New("error resizing console pty is closed")
+	}
+
+	if _, _, err := syscall.Syscall(syscall.SYS_IOCTL, r.pty.Fd(), uintptr(unix.TIOCSWINSZ), uintptr(unsafe.Pointer(&consoleSize{Height: height, Width: width}))); err != 0 {
```

Comment:
>Syscall [](start = 25, length = 7)

Probably you should make a standalone function in tty.go that calls this IOCTL that can be used or tested separately from the relay.